src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseTypeDefinition.java

```diff
@@ -168,4 +174,27 @@ public boolean isCaseFieldACollection(String caseFieldId) {
     public Optional<CaseFieldDefinition> getCaseField(String caseFieldId) {
         return caseFieldDefinitions.stream().filter(caseField -> caseField.getId().equalsIgnoreCase(caseFieldId)).findFirst();
     }
+
+    @JsonIgnore
+    public Optional<CommonField> getCommonFieldByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            return Optional.empty();
+        }
+        List<String> pathElements = Arrays.stream(path.trim().split("\\.")).collect(toList());
```

Comment:
lots of code emerging more and more to deal with path manipulation. 2 examples in this class, here another: 
https://github.com/hmcts/ccd-data-store-api/blob/0fc75ab4364e95f78271f62bb04322f94b406a4f/src/main/java/uk/gov/hmcts/ccd/domain/model/definition/SearchResultField.java#L61

Can we gather all path related code into a new class `CaseFieldPathUtils` class please